### PR TITLE
fix: Change re-export syntax to expose default explicitly or export all

### PR DESCRIPTION
**Pull request by jsJunky**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (`Closes #XXXX`): Closes 
#### Description

Intellij was warning about the re-export syntax used so changed to either export all exposed functions or explicitly export the default.

Syntax for re-exporting found [here](http://exploringjs.com/es6/ch_modules.html)

fix: export the individual default export

fix: remove trailing comma syntax error

fix: re-export all actions instead of default

### CODE REVIEW

File: common/app/redux/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export { default as reducer } from './reducer';
-export { default as actions } from './actions';
+export * from './actions';
```
Comment (BerkeleyTrue) [inline common/app/redux/index.js]: This changes the export from `import  { actions } from 'x'` to `import  { individualAction } from 'x';`

File: common/app/components/Footer/index.js [hunk 2]
```diff
@@ -1 +1 @@
-export default from './Footer.jsx';
```
Comment (BerkeleyTrue) [inline common/app/components/Footer/index.js]: This is valid syntax when using https://github.com/leebyron/ecmascript-export-default-from

File: common/app/routes/Hikes/redux/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-export actions from './actions';
-export reducer from './reducer';
-export types from './types';
```
Comment (BerkeleyTrue) [inline common/app/routes/Hikes/redux/index.js]: These two lines are valid syntax.

see: https://github.com/leebyron/ecmascript-export-default-from

File: common/app/routes/Hikes/redux/index.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
+export * from './actions';
```
Comment (BerkeleyTrue) [inline common/app/routes/Hikes/redux/index.js]: This should be `export * as actions from './actions'`

see: https://github.com/leebyron/ecmascript-export-default-from

File: common/app/components/Nav/index.js [hunk 5]
```diff
@@ -1 +1 @@
-export default from './Nav.jsx';
```
Comment (BerkeleyTrue) [inline common/app/components/Nav/index.js]: This is valid syntax:

see https://github.com/leebyron/ecmascript-export-default-from

Comment (jsJunky) [inline common/app/components/Nav/index.js]: Ah I didn't know about that proposal! I like the simplification. I'll remove these changes and update the other two to be * as v syntax. I don't know the proposal structure how long does a stage 1 proposal take to get through?

**Comment by BerkeleyTrue** [comment]
Closing as abandoned. Feel free to re-open when ready.

Happy Coding!

**Pull request closed without merging**